### DISCUSSION TITLE
Run docker inside sandbox

### DISCUSSION THREAD

**Discussion by gagara** [netblue30/firejail#4582] [Q&A] [answered]
I'm trying to run docker inside sandbox.
Now it fails as `/sys` is modified and seems like `/sys/fs/cgroups` are not available.

My question is following:
Is there any way to run sandbox with untouched mount namespace? Something like `whitelist /` or `chroot=/`?

**Comment by rusty-snake** [comment, marked as answer]
Stacking of containers/sandboxes (it's the same at the end) is always difficult however theoretically it is possible.

> Is there any way to run sandbox with untouched mount namespace?

Firejail does not have such an option yet. Anyway what's the goal of a "sandbox" that allows anything?

>  seems like /sys/fs/cgroups are not available.

`noblacklist /sys/fs`
See also [noprofile.profile](https://gist.github.com/rusty-snake/bb234cb3e50e1e4e7429f29a7931cc72).

**Reply by gagara** [reply]
Actually docker-in-docker is rather wide spread solution. Works well.
My goal is following: I need docker with access to private repository behind VPN. However rest of the system should bypass vpn. Possible solution is to run vpn client and docker inside one network namespace. This works well when i create manually network namespace. Mount namespace i need only to bind mount /etc/resolv.conf (unfortunatelly also not easy in firejail) as vpn clients usualy update it.
If we have option to not use net NS why not have option to not use mount NS? or rather use it but w/o any tweaks  by default.
Thanks for noprofile. Didnt know about it. Will give it a try.
Thx

**Comment by gagara** [comment]
worked with noprofile setting. Thanks a lot
